src/main/java/org/openstreetmap/atlas/tags/annotations/extraction/LengthExtractor.java

```diff
@@ -0,0 +1,87 @@
+package org.openstreetmap.atlas.tags.annotations.extraction;
+
+import java.util.Optional;
+
+import org.openstreetmap.atlas.tags.annotations.Tag;
+import org.openstreetmap.atlas.tags.annotations.validation.LengthValidator;
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * Extracts a {@link Distance} from a value. Can be used on tag values such as
+ * {@link org.openstreetmap.atlas.tags.WidthTag}.
+ *
+ * @author bbreithaupt
+ */
+public class LengthExtractor implements TagExtractor
+{
+    private static final LengthValidator VALIDATOR = new LengthValidator();
+    private static final String METERS_SUFFIX = " m";
```

Comment:
Might be worth having a reusable unit enum class (if it doesn't already exist)

Reply:
I created an enum set and a couple of constants in the Distance class. I put them there to match the speed unit abbreviations that are defined in the Speed class.